src/lxml/tests/test_xmlschema.py

```diff
@@ -80,7 +80,7 @@ def test_xmlschema_error_log_path(self):
         schema.validate(tree)
         tree_path = tree.getpath(tree.findall('b')[1])
         error_path = schema.error_log[0].path
-        self.assertTrue(tree_path == error_path)
+        self.assertTrue(error_path is None or tree_path == error_path)
```

Comment:
Actually, looking back at http://www.xmlsoft.org/html/libxml-xmlerror.html#xmlError I think that was a mistake. I had read the comment earlier about the possibility that the parser context not be available ("the parser context if available"), and I mis-remembered that as applying to the xmlNode pointer as well, but looking back I see that there is no such caveat. We do know that there are xmlError structs floating around which have a NULL node pointer (we're creating some ourselves), but a good case could be made that the test should just assume that the pointer will always be non-NULL for schema validation errors. As you can see, that's how I originally wrote the test. Should I back out that change? If the validity of our assumption ever changes at some point in the future we can modify the test again.